extensions/mssql/src/utils.ts

```diff
@@ -31,6 +42,90 @@ export function getAppDataPath() {
 	}
 }
 
+/**
+ * Returns the credential folder path on the user's system
+ */
+export async function removeCredentialFile(): Promise<void> {
+	const home = os.homedir();
+	const credentialPath = path.join(home, '.sqlsecrets');
+	await fs.rmdir(credentialPath, { recursive: true });
+}
+
+/**
+ *
+ * @returns Whether the current OS is linux or not
+ */
+export function isLinux(): boolean {
+	return os.platform() === 'linux';
+}
+
+/**
+ * This function returns whether the native credential management system
+ * should be used instead of tools service
+ */
+export function keytarCredentialsEnabled(): boolean {
+	const linux: boolean = isLinux();
+	const useKeytarCredential: boolean = getConfigKeytar();
+	return linux && useKeytarCredential;
+}
+
+/**
+ * Gets the prefix for a formatted credential ID
+ */
+
+function getConnectionInfoId(connectionProfile: azdata.connection.ConnectionProfile): string {
+	let idNames = ['authenticationType', 'database', 'server', 'user', 'applicationName'];
+
+	//Sort to make sure using names in the same order every time otherwise the ids would be different
+	idNames.sort();
+
+	let idValues: string[] = [];
+	for (let index = 0; index < idNames.length; index++) {
+		let value = connectionProfile.options[idNames[index]!];
+		value = value ? value : '';
+		idValues.push(`${idNames[index]}${CRED_NAME_VALUE_SEPARATOR}${value}`);
+	}
+	return CRED_PROVIDER_NAME + CRED_NAME_VALUE_SEPARATOR + connectionProfile.providerId +
+		CRED_SEPARATOR + idValues.join(CRED_SEPARATOR);
+}
+
+/**
+ * Creates a formatted credential usable for uniquely identifying a SQL Connection.
+ * This string can be decoded but is not optimized for this.
+ * @param connectionProfile connection profile - require
+ * @param itemType type of the item (MRU or Profile) - optional
+ * @returns formatted string with server, DB and username
+ */
+export function formatCredentialId(connectionProfile: azdata.connection.ConnectionProfile): string {
+	const cred: string[] = [CRED_PREFIX];
+	cred.push(CRED_ITEMTYPE_PREFIX.concat(CRED_PROFILE_USER));
+	cred.push(CRED_ID_PREFIX.concat(getConnectionInfoId(connectionProfile)));
+	return cred.join(CRED_SEPARATOR);
+}
+
+/**
+ * This function returns whether the user wants to delete the old credentials
+ * from their Linux system when migrating to keytar
+ * It's true by default, and the system tries to migrate if true, else skips
+ * since it can only be false if the user set it or the migration is done
+ * @returns
+ */
+export function migrateLinuxCredentials(): boolean {
+	const linux: boolean = isLinux();
+	const migrateLinuxCredentials: boolean = getConfigMigrateLinuxCredential();
```

Comment:
You should use a memento or something else for this instead - it would seem weird to have this show up in the settings configuration (especially since this is only applicable for linux systems)